Plugins/Mute/MuteManager.cs

```diff
@@ -1,21 +1,58 @@
 ﻿using SharedLibraryCore;
+using SharedLibraryCore.Database.Models;
 
 namespace Mute;
 
 public class MuteManager
 {
-    public async Task<bool> Mute(GameEvent gameEvent)
+    private List<MuteStateMeta> _muteList = null!;
+    public bool HasMutes => _muteList.Any();
+    public IEnumerable<MuteStateMeta> Mutes => _muteList;
+    public async void InitialiseMuteList() => _muteList = await Plugin.DataManager.ReadMuteList();
+
+    public async Task<bool> Mute(Server server, EFClient origin, EFClient target, DateTime? dateTime, string reason)
+    {
+        if ((await Plugin.DataManager.ReadPersistentData(target)).MuteState == MuteState.Muted) return false;
```

Comment:
following the previous comment, that will make the result of read potentially null, so handle appropriately